server/src/main/java/com/cloud/network/NetworkServiceImpl.java

```diff
@@ -1788,6 +1788,14 @@ public Network createGuestNetwork(CreateNetworkCmd cmd) throws InsufficientCapac
 
         checkNetworkDns(ipv6, ntwkOff, vpcId, ip4Dns1, ip4Dns2, ip6Dns1, ip6Dns2);
 
+        if (vpcId != null && VpcManager.VpcTierNamePrepend.value()) {
+            final String delimiter = VpcManager.VpcTierNamePrependDelimiter.value();
+            Vpc vpc = _vpcDao.findById(vpcId);
+            if (vpc != null) {
+                name = vpc.getName() + delimiter + name;
+            }
+        }
```

Comment:
can this be a method called at the moment `name` is being set?

Reply:
Done. please check.